byway/src/main/java/com/google/sps/servlets/StopsServlet.java

```diff
@@ -26,26 +37,39 @@
 /** Servlet that handles data for stops */
 @WebServlet("/api/stop")
 public final class StopsServlet extends HttpServlet {
-  @GuardedBy("this")
-  private ArrayList<String> stops = new ArrayList<String>();
+  private static final Type ARRAYLIST_STRING = new TypeToken<ArrayList<String>>() {}.getType();
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
 
   private final Gson gson = new Gson();
-  // Implement datastore in routepage branch
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final Key key = KeyFactory.createKey(Trip.KIND, 1234);
 
+  /* Passes saved destinations stops (if any) to be shown in the schedule panel */
   @Override
-  public synchronized void doGet(HttpServletRequest request, HttpServletResponse response)
-      throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(stops);
-
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> stops = Collections.emptyList();
+    try {
+      Entity entity = datastore.get(key);
+      stops = (ArrayList<String>) entity.getProperty("destinations");
+    } catch (EntityNotFoundException e) {
+      logger.atInfo().withCause(e).log(
+          "Could not retrieve Entity for Trip with key %s while trying to get the stops", key);
+    }
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(gson.toJson(stops));
   }
 
+  /* Modifies the destinations array of Trip entity in datastore */
   @Override
-  public synchronized void doPost(HttpServletRequest request, HttpServletResponse response)
-      throws IOException {
-    String stop = request.getParameter("text");
-    stops.add(stop);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<String> stops = gson.fromJson(request.getReader(), ARRAYLIST_STRING);
+    try {
+      Entity entity = datastore.get(key);
```

Comment:
When you make a `try` block, try to make it as narrowly scoped as you can. That way you don't accidentally catch errors from parts of the code that you didn't mean to handle. This is important for debugging, because if you catch errors you didn't intend to you may not realize what the actual source of an error is, or you can introduce bugs, for example if you handle `EntityNotFound` by creating the requested trip entity, but it turns out you actually caught `EntityNotFound` from code that was trying to get the User, you might overwrite an existing trip by accident.

Reply:
Forgot to comment, done.